File: cpp_rust_ch05/main.cpp
#include <iostream>

void by_value(int);
void by_reference(int&);

int main()
{
    int age{45};
    by_value(age);

    by_reference(age);

    // by_value(age);
}


void by_value(int age)
{
    age *= 5;
    std::cout << "The new age is " << age << std::endl;
}

void by_reference(int& age)
{
    age *= -3;

    std::cout << "The new age is " << age << std::endl;
}

File: cpp_rust_ch05/src/main.rs
fn main() {
    let age = 45;
    by_reference(&age);

    by_value(age);
}

fn by_reference(x: &i32) {
    println!("{}", x); // immutable borrow
}

fn by_value(x: &mut i32) {
    *x += 1; // mutable borrow
}
